bokehjs/src/coffee/models/ranges/data_range1d.coffee

```diff
@@ -125,9 +136,17 @@ export class DataRange1d extends DataRange
     [start, end] = @_compute_range(min, max)
 
     if @_initial_start?
-      start = @_initial_start
+      if @_mapper_type == "log"
+        if @_initial_start > 0
+          start = @_initial_start
+      else
+        start = @_initial_start
     if @_initial_end?
-      end = @_initial_end
+      if @_mapper_type == "log"
+        if @_initial_end > 0
+          end = @_initial_end
+      else
+        end = @_initial_end
```

Comment:
Well MPL at least has `symlog` for which will linearize around zero and allow for negative values, but the regular `log` mode has both `clip` and `mask` options I thought, and in particular, that only applies to *values* AFAICT. You can still set range bounds < 0 if you like, and they are honored as best I can tell. If we are going to prevent *ranges* from extending to negative in the presence of a log mapper, that still seems worth a warning to me. Alternatively we can just let ranges extend below zero, but refuse to map negative values like MPL does.